scripts/build_glpk.py

```diff
@@ -1,44 +1,33 @@
 # taken from https://github.com/opencobra/cobrapy/blob/devel/appveyor/build_glpk.py
 
 import os
-import sys
-# import hashlib
 import tarfile
 import struct
 import shutil
-import setuptools.msvc
 import urllib.request as urllib2
 import subprocess
 
+GUESS_VCVARS = (
+    "C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\"
+    "VC\\Auxiliary\\Build\\vcvarsall.bat"
+)
+
 # these need to be set to the latest glpk version
 glpk_version = os.getenv('NEW_GLPK_VERSION')
-# glpk_md5 = "eda7965907f6919ffc69801646f13c3e"
 
 glpk_build_dir = "glpk_build/glpk-%s" % glpk_version
 url = "http://ftp.gnu.org/gnu/glpk/glpk-%s.tar.gz" % glpk_version
 bitness = struct.calcsize("P") * 8
 arch = "x86_amd64" if bitness == 64 else "x86"
 
-
-# def md5(fname):
-#     hash = hashlib.md5()
-#     with open(fname, "rb") as f:
-#         for chunk in iter(lambda: f.read(4096), b""):
-#             hash.update(chunk)
-#     return hash.hexdigest()
-
-
-def get_vcvarsall_cmd():
-    py_ver = sys.version_info
-    if py_ver.major == 3 and py_ver.minor >= 5:
-        vc_ver = 14
-    elif py_ver.major == 3 and py_ver.minor >= 3:
-        vc_ver = 10
-    else:
-        vc_ver = 9
-    vc_path = setuptools.msvc.msvc9_find_vcvarsall(vc_ver)
-    assert vc_path is not None
-    return '"%s" %s' % (vc_path, " amd64" if bitness == 64 else "")
+def find_vcvarsall():
+    if os.path.isfile(GUESS_VCVARS):
+        return(GUESS_VCVARS)
+    for root, dirs, files in os.walk("C:\\Program Files\\Microsoft Visual Studio\\"):
+        for f in files:
+            if f == "vcvarsall.bat":
+                return(os.path.join(root, *dirs, f))
```

Comment:
This does not make sense to me, the first element of the triple (`root`) is the current directory where a file is placed. Should be `join(root, f)`, I think.

https://docs.python.org/3/library/os.html#os.walk